libr/bin/p/bin_ninds.c

```diff
@@ -33,13 +33,15 @@ static bool load_bytes(RBinFile *bf, void **bin_obj, const ut8 *buf, ut64 sz, ut
 }
 
 static bool load(RBinFile *bf) {
-	const ut8 *bytes = bf? r_buf_buffer (bf->buf): NULL;
-	ut64 sz = bf? r_buf_size (bf->buf): 0;
 	if (!bf || !bf->o) {
 		return false;
 	}
+	ut64 sz;
+	ut8 *bytes = r_buf_buffer (bf->buf, &sz);
 	load_bytes (bf, &bf->o->bin_obj, bytes, sz, bf->o->loadaddr, bf->sdb);
-	return check_bytes (bytes, sz);
+	bool res = check_bytes (bytes, sz);
+	free (bytes);
```

Comment:
wtf? r_buf_bnuffer is now returning a new buffer with all the bytes? this is gonna be a huge pain for big files

Reply:
yes :/ it's the only way.
`r_buf_buffer` should not be used! Consider that `r_buf_buffer` could be used only for "bytes buffer" and the buffer was already entirely allocated in memory. The problem is that r_buf_buffer could not be used for "file buffer" or other type of buffers. These changes allow us to implement `r_buf_buffer` for any kind of buffer, but of course you have to be very careful. Again, if you use `r_buf_buffer` you are defeating the very purpose of RBuffer, because you are asking to have the entire buffer allocated in memory (this was true before and it's still true. The only difference is that now you have a new allocated buffer, so that the method can be used for other types of buffers as well)